core/src/main/scala/org/apache/spark/scheduler/ExecutorLossReason.scala

```diff
@@ -54,10 +54,13 @@ private [spark] object LossReasonPending extends ExecutorLossReason("Pending los
 /**
  * @param _message human readable loss reason
  * @param workerLost whether the worker is confirmed lost too (i.e. including shuffle service)
+ *                   Used to know whether the shuffle data is lost too.
+ * @param causedByApp whether the loss of the slave is the fault of the running app. (assumed true
+ *                    by default unless known explicitly otherwise)
  */
 private[spark]
-case class SlaveLost(_message: String = "Slave lost", workerLost: Boolean = false)
-  extends ExecutorLossReason(_message)
+case class SlaveLost(_message: String = "Slave lost", workerLost: Boolean = false,
+                     causedByApp: Boolean = true) extends ExecutorLossReason(_message)
```

Comment:
nit:
```suggestion
case class SlaveLost(
    _message: String = "Slave lost",
    workerLost: Boolean = false,
    causedByApp: Boolean = true)
  extends ExecutorLossReason(_message)
```

Reply:
BTW, `causedByApp` is not used at all?

Reply:
Its used a pattern match in TaskSetManager.scala, around line 984, maybe that is why it wasn't greppable.